### ISSUE TITLE
Convert to .net 4.0

### ISSUE THREAD

**Issue by riccardone** [Azure/amqpnetlite#49]
I'm trying to convert the Amqp.net project (ContainerHost class and all dependencies) to .net 4.0. I will open a pull request if I can complete this task. Are there some 4.5 features that are you using in it and that you think that are difficult to substitute?
Thanks.

**Comment by xinchen10** [comment]
The following are not available in 4.0. 
1. async/await - probably can use Microsoft.Bcl.Async without changing code
2. System.Net.WebSockets.WebSocket - need to be excluded

**Comment by riccardone** [comment]
I'm working on this.

**Comment by riccardone** [comment]
Hi Xinchen,
I started with this but then I found it too difficult. I tried to use an old version of your code but no luck and I can't use the Microsoft Bcl that you've mentioned.
Instead of convert amqp.net to 4.0 I upgraded the project where I'm trying to implement an amqp listner to the latest .net and then reference amqp.net as is.
I'm saying this in case you want to close this ticket if is not relevant for anyone else.

Regards
Riccardo
